cli/pkg/artifact_builder/artifact_builder.go

```diff
@@ -219,23 +233,58 @@ func (ab ArtifactBuilder) RegistryLogin(ctx context.Context) error {
 	if err != nil {
 		return errors.Wrap(err, "artifact builder configuration is incomplete")
 	}
-
 	ecrAuthorizationToken, err := ab.backend.ECRGetAuthorizationToken(ctx)
 	if err != nil {
 		return err
 	}
 
 	args := []string{"login", "--username", ecrAuthorizationToken.Username, "--password", ecrAuthorizationToken.Password, ecrAuthorizationToken.ProxyEndpoint}
 
-	docker, err := ab.config.executor.LookPath("docker")
+	docker, err := ab.config.Executor.LookPath("docker")
 	if err != nil {
 		return errors.Wrap(err, "could not find docker in path")
 	}
 	cmd := exec.CommandContext(ctx, docker, args...)
-	err = ab.config.executor.Run(cmd)
+	err = ab.config.Executor.Run(cmd)
 	return errors.Wrap(err, "registry login failed")
 }
 
+func (ab ArtifactBuilder) GetECRsForServices(ctx context.Context) (map[string]*config.RegistryConfig, error) {
+	var (
+		maxResults int32   = 50
+		nextToken  *string = nil
+		repos              = []ecrtypes.Repository{}
+	)
+	for {
+		r, err := ab.backend.GetECRClient().DescribeRepositories(ctx, &ecr.DescribeRepositoriesInput{
+			MaxResults: &maxResults,
+			NextToken:  nextToken,
+		})
+		if err != nil {
+			return nil, errors.Wrap(err, "unable to describe ECR repos")
+		}
+		if len(r.Repositories) == 0 {
+			break
+		}
+		repos = append(repos, r.Repositories...)
+		if r.NextToken == nil {
+			break
+		}
+		nextToken = r.NextToken
+	}
+	ecrs := map[string]*config.RegistryConfig{}
+	for _, service := range ab.backend.Conf().HappyConfig.GetServices() {
+		for _, repo := range repos {
+			if repo.RepositoryName != nil &&
+				repo.RepositoryUri != nil &&
+				*repo.RepositoryName == fmt.Sprintf("%s-%s-%s", ab.config.StackName, ab.config.env, service) {
```

Comment:
I propose we use `/` as a separator character for readability. We can have conflicting repo names, as we often use dashes in our naming. `jheath-prod-prod-frontend` could be interpreted in several different ways, why `jheath-prod/prod/frontend` won't. I believe we already disallow `/` as a part of the stack name, so no duplicity there.

Reply:
I could also run the terraform targeted apply each time and have terraform manage that.

Reply:
o actually, I think I can use the state of the targeted apply to get these. I don't think I need to make an API call.